osquery/utils/system/linux/proc/proc.cpp

```diff
@@ -51,5 +57,81 @@ std::string cmdline(pid_t const pid) {
   return content;
 }
 
+static std::string makeNSPath(const char* pid_or_path) {
+  std::string nspath = pid_or_path;
+  if (nspath.empty()) {
+    return nspath;
+  }
+  if (nspath[0] == '/') {
+    return nspath;
+  }
+  nspath = kProc + "/";
+  nspath += pid_or_path;
+  nspath += kMountNamespace;
+  return nspath;
+}
+
+/*
+ * Use SYS_setns syscall to set namespace.
+ * NOTE: setns() function not added until glibc 2.14
+ */
+Status setLinuxNamespace(const char* cpath, std::string& feedback) {
+  static int origfd = 0;
+  static std::string lastns;
+  std::string nspath = makeNSPath(cpath);
+
+  // need to keep handle to original namespace
+
+  if (origfd <= 0) {
+    std::string original_mnt_path =
+        kProc + "/" + std::to_string(getpid()) + kMountNamespace;
+
+    origfd = open(original_mnt_path.c_str(), O_RDONLY);
+    if (origfd <= 0) {
+      feedback +=
+          "Unable to open original namespace descriptor: " + original_mnt_path +
+          "\n";
+    }
+  }
+
+  // if in a namespace, restore original
+
+  if (!lastns.empty()) {
+    lastns.clear();
+
+    // restore original namespace
```

Comment:
Providing a blank value resets osqueryi to its default namespace, this should be documented in the help output for the flag, and in the wiki doc I think.